wremnants/syst_tools.py

```diff
@@ -361,6 +361,17 @@ def add_pdf_hists(results, df, dataset, axes, cols, pdfs, base_name="nominal", a
         else:
             pdfHist = df.HistoBoost(pdfHistName, axes, [*cols, tensorName], tensor_axes=[pdf_ax], storage=hist.storage.Double())
             alphaSHist = df.HistoBoost(alphaSHistName, axes, [*cols, tensorASName], tensor_axes=[as_ax], storage=hist.storage.Double())
+
+            if propagateToHelicity:
+                df=df.Define("unity","1.")
+                pdfhelper = ROOT.wrem.makeHelicityMomentPdfTensor[npdf]()
+                df = df.Define(f"helicity_moments_{tensorName}_tensor", pdfhelper, ["csSineCosThetaPhi", f"{tensorName}", "unity"])
```

Comment:
So actually @emanca just to confirm the reason why "unity" is passed as the nominal weight here is that the pdf tensor already has the nominal weight taken into account?

Reply:
This option is actually broken for the moment because there is a mismatch in the length of the alphaS tensor vs the corresponding tensor axis (2 vs 3).  I propose we merge as is and fix it in a follow-up PR.

Reply:
I was updating the alphaS tensor to be length 3 for my recent changes. Will that break or fix things?

Reply:
Updating it to 3 will fix things in that case.